src/index.js

```diff
@@ -62,11 +62,18 @@ export default function testStorySnapshots (options = {}) {
           it(story.name, () => {
             const context = { kind: group.kind, story: story.name }
             const renderedStory = story.render(context)
-            const tree = renderer.create(renderedStory).toJSON()
+            const tree = renderer.create(renderedStory, getTestRendererOptions(story)).toJSON()
             expect(tree).toMatchSnapshot()
           })
         }
       })
     })
   }
 }
+
+const getRenderOptions = story => (story && story.render && story.render.options) ? story.render.options : {}
+
+const getTestRendererOptions = (
+  story,
+  options = getRenderOptions(story)
+) => (options.createNodeMock) ? { createNodeMock: options.createNodeMock } : null
```

Comment:
Perhaps plain JS functions would be better for [hoisting](https://scotch.io/tutorials/understanding-hoisting-in-javascript) and/or readability reasons.